pgmpy/base/simple_causal_model.py

```diff
@@ -0,0 +1,140 @@
+    ... )
+    >>> model2.nodes()
+    ['Var_1', 'Var_2', 'Var_3', 'Var_4', 'Var_5']
+    >>> model2.edges()
+    OutEdgeView([
+            ('Var_1', 'Var_2'), ('Var_3', 'Var_1'), ('Var_3', 'Var_2'),
+            ('Var_5', 'Var_1'), ('Var_1', 'Var_4'), ('Var_4', 'Var_2')])
+    """
+
+    @staticmethod
+    def _to_list(var):
+        if var is None:
+            return []
+        elif isinstance(var, str):
+            return [var]
+        elif isinstance(var, int):
+            return [f"Var_{var}"]
+        return list(var)
+
+    def __init__(
+        self,
+        exposures: Union[str, int, Iterable[Union[str, int]]],
+        outcomes: Union[str, int, Iterable[Union[str, int]]],
+        covariates: Optional[Union[str, int, Iterable[Union[str, int]]]] = None,
+        mediators: Optional[Union[str, int, Iterable[Union[str, int]]]] = None,
+        instruments: Optional[Union[str, int, Iterable[Union[str, int]]]] = None,
+        latents: Optional[Iterable[str]] = None,
+    ):
+        exposures = self._to_list(exposures)
+        outcomes = self._to_list(outcomes)
+        covariates = self._to_list(covariates)
+        mediators = self._to_list(mediators)
+        instruments = self._to_list(instruments)
+        latents = list(latents) if latents is not None else []
+
+        edges = []
+
+        # Add edges from exposures to outcomes
+        edges += [(exp, out) for exp in exposures for out in outcomes]
+
+        # Add edges from covariates to exposures and outcomes
+        edges += [(cov, exp) for cov in covariates for exp in exposures]
+        edges += [(cov, out) for cov in covariates for out in outcomes]
+
+        # Add edges from instruments to exposures
+        edges += [(inst, exp) for inst in instruments for exp in exposures]
+
+        # Add edges from exposures to mediators and mediators to outcomes
+        edges += [(exp, med) for exp in exposures for med in mediators]
+        edges += [(med, out) for med in mediators for out in outcomes]
+
+        super().__init__(edges)
+
+        # Add latent nodes to the graph and set the latents attribute.
+        latents_set = set(latents) if latents else set()
+        if latents_set:
+            self.add_nodes_from(latents_set)
+        self.latents = latents_set
+
+        self.variable_roles = {
+            "exposure": set(exposures),
+            "outcome": set(outcomes),
+            "covariate": set(covariates),
+            "mediator": set(mediators),
+            "instrument": set(instruments),
+        }
+
+    def get_role(self, role):
+        """
+        Returns the set of variables for a given role.
+
+        Parameters
+        ----------
+        role: str
+            The role to query (e.g., "exposure", "outcome", "covariate", "mediator", "instrument").
+
+        Returns
+        -------
+        set
+            A set of variable names associated with the role.
+        """
+        return self.variable_roles.get(role, set())
```

Comment:
This class inherits `DAG` class which inherits `_GraphRolesMixin`. Since `_GraphRolesMixin` already has `get_role` method, it should automatically work with this class as well.